From ddde5ccb145d5848cad388078a33f40517461b62 Mon Sep 17 00:00:00 2001
From: gli41 <genshen.li@intel.com>
Date: Thu, 21 Jul 2016 21:35:27 +0800
Subject: [PATCH 1/3] Make dm-verity-target.c consistent with kernel common
 git.

The dm verity parameter parsing need to be changed to take into account
optional parameters such as FEC, otherwise FEC won't work as expected.

Change-Id: I0614c8b2a8e0c0e41621daebb4168962d28a0c37
Signed-off-by: gli41 <genshen.li@intel.com>
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-35882
Reviewed-on: https://android.intel.com:443/525187
---
 drivers/md/dm-verity-target.c | 19 +++++++++++--------
 1 file changed, 11 insertions(+), 8 deletions(-)

diff --git a/drivers/md/dm-verity-target.c b/drivers/md/dm-verity-target.c
index 7d95ca6..c0331eb 100644
--- a/drivers/md/dm-verity-target.c
+++ b/drivers/md/dm-verity-target.c
@@ -837,6 +837,7 @@ static int verity_parse_opt_args(struct dm_arg_set *as, struct dm_verity *v)
 int verity_ctr(struct dm_target *ti, unsigned argc, char **argv)
 {
 	struct dm_verity *v;
+	struct dm_arg_set as;
 	unsigned int num;
 	unsigned long long num_ll;
 	int r;
@@ -984,15 +985,17 @@ int verity_ctr(struct dm_target *ti, unsigned argc, char **argv)
 		}
 	}
 
-	if (argc > 10) {
-		if (sscanf(argv[10], "%d%c", &num, &dummy) != 1 ||
-		    num < DM_VERITY_MODE_EIO ||
-		    num > DM_VERITY_MODE_RESTART) {
-			ti->error = "Invalid mode";
-			r = -EINVAL;
+	argv += 10;
+	argc -= 10;
+
+	/* Optional parameters */
+	if (argc) {
+		as.argc = argc;
+		as.argv = argv;
+
+		r = verity_parse_opt_args(&as, v);
+		if (r < 0)
 			goto bad;
- 		}
-		v->mode = num;
 	}
 
 	v->hash_per_block_bits =
-- 
1.9.1

